src/viam/robot/client.py

```diff
@@ -101,6 +102,32 @@ class Options:
         Whether sessions are disabled
         """
 
+        @classmethod
+        def with_api_key(cls, api_key: str, api_key_id: str) -> Self:
+            """
+            Create dial options with an API key for credentials and default values for other arguments.
+
+            Args:
+                api_key (str): your API key
+                api_key_id (str): your API key ID
+
+            Raises:
+                ValueError: Raised if the `api_key_id` is not a valid uuid
+
+            Returns:
+                Self: the RobotClient options
+            """
+            key_id_regex = re.compile(r"^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$")
+            match = key_id_regex.match(api_key_id)
+            if not match:
+                raise ValueError(f"{api_key_id} is not a valid UUID")
```

Comment:
This might be easier using the UUID library in python:

```python3
import uuid

...

try:
    uuid.UUID(api_key_id)
except ValueError:
    raise ValueError(f"{api_key_id} is not a valid UUID")
```

Reply:
ooh thanks for the tip, didn't know about the `UUID` library!